libraries/surplus/src/advanced-tests.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import S from 's-js';
+import * as Surplus from "surplus";
+import expect from "expect";
+import {
+  ComponentWithoutChildren,
+  ComponentWithChildren,
+  ComponentWithChildrenRerender,
+  ComponentWithDifferentViews,
+  ComponentWithProperties,
+  ComponentWithUnregistered,
+  ComponentWithImperativeEvent,
+  ComponentWithDeclarativeEvent
+} from "./components";
+
+
+//let _it = it;
+//it = function it(title, fn) { return _it.call(this, title, () => S.root(fn.bind(this))); }
+
+describe("advanced support", function() {
+
+  describe("attributes and properties", function() {
+    it("will pass array data as a property", function() {
+      S.root(() => {
+        this.weight = 2;
```

Comment:
for consistency would you mind moving all calls to `this.weight` out of this function.

Reply:
Ah!  This explains why my attempt to wrap `it()` didn't work -- the weight calculations weren't passing through.  Will fix.